source/telemetry/telemetry_logging/logging_configuration.py

```diff
@@ -11,50 +11,102 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 import contextlib
 import logging
 import os
-from typing import Any, Iterator
+from typing import Any, Iterator, Tuple, Type, Optional
+from pydantic_settings import (
+    BaseSettings,
+    CliSettingsSource,
+    PydanticBaseSettingsSource,
+    SettingsConfigDict,
+)
+from pydantic import Field
+from uuid import UUID
 from azure.monitor.opentelemetry import configure_azure_monitor
 from opentelemetry import trace
 from opentelemetry.trace import Span, Tracer
 
+
 DEFAULT_LOG_FORMAT: str = "%(asctime)s %(levelname)s %(name)s: %(message)s"
 DEFAULT_LOG_LEVEL: int = logging.INFO
 _EXTRAS: dict[str, Any] = {}
 _IS_INSTRUMENTED: bool = False
 _TRACER: Tracer | None = None
 _TRACER_NAME: str
+_LOGGING_CONFIGURED: bool = False  # Flag to track if logging is configured
+
+
+def set_logging_configured(configured: bool) -> None:
+    """Sets the global flag indicating logging has been configured."""
+    global _LOGGING_CONFIGURED
+    _LOGGING_CONFIGURED = configured
+
+
+def get_logging_configured() -> bool:
+    """Returns the current logging configuration state."""
+    return _LOGGING_CONFIGURED
+
+
+class LoggingSettings(BaseSettings):
+    """
+    LoggingSettings class uses Pydantic BaseSettings to configure and validate parameters.
+    Parameters can come from both runtime (CLI) or from environment variables.
+    The priority is CLI parameters first and then environment variables.
+    """
+
+    cloud_role_name: str
+    applicationinsights_connection_string: str = Field(
+        alias="applicationinsights-connection-string", default=None
+    )
+    subsystem: str
+    orchestration_instance_id: UUID = Field(alias="orchestration-instance-id")
```

Comment:
There is no orchestration_instance_id in geh-settlement-reports as creation of the reports are not initiated by an orchestration.